evm/src/cpu/kernel/interpreter.rs

```diff
@@ -513,6 +577,43 @@ impl<'a> Interpreter<'a> {
         self.push_bool(x > y);
     }
 
+    fn run_slt(&mut self) {
+        let x = self.pop();
+        let y = self.pop();
+        self.push_bool(signed_cmp(x, y) == Ordering::Less);
+    }
+
+    fn run_sgt(&mut self) {
+        let x = self.pop();
+        let y = self.pop();
+        self.push_bool(signed_cmp(x, y) == Ordering::Greater);
+    }
+
+    fn run_signextend(&mut self) {
+        let n = self.pop();
+        let x = self.pop();
+        if n > U256::from(31) {
+            self.push(x);
+        } else {
+            let n = n.low_u64() as usize;
+            let num_bytes_prepend = 31 - n;
+
+            let mut x_bytes = [0u8; 32];
+            x.to_big_endian(&mut x_bytes);
+            let mut x_bytes = x_bytes[num_bytes_prepend..].to_vec();
+            let sign_bit = x_bytes[0] >> 7;
+
+            self.push(if sign_bit == 0 {
+                x_bytes.extend_from_slice(&vec![0; num_bytes_prepend]);
```

Comment:
It looks like here the sign bits are appended after x, should it be prepended before?

Reply:
Yes you're right, I've messed up the ordering here. Sorry about that, fixed!